container.go

```diff
@@ -410,6 +448,8 @@ func (c *Container) start() error {
 		return err
 	}
 
+	c.storeMounts()
```

Comment:
I think you should remove that and store the mount points from the agent itself.